### DISCUSSION TITLE
Multi-position circuits

### DISCUSSION THREAD

**Discussion by alehaa** [netbox-community/netbox#16917] [Help Wanted!]
Has anyone tried to model infrastructure with more than one position (e.g. multi-fiber cable)? I can't find a good way to do this with NetBox out of the box.

Example:
* We have two sites.
* Both are connected to a central hub operated by someone else using cables with 96 positions each. The cables are also owned by the hub operator.
* 12 positions are interconnected between the two cables at the hub, others go to our ISP, other peers, etc.
* It's convenient to know which port goes where at a site, so I added a patch panel as a device, using front and rear ports. However, since only one cable leaves the building, I've mapped all the front ports to a single rear port with 96 positions and connected it to a circuit termination. 

This leaves me with the problem that I can only connect the entire circuit to another circuit or network. To interconnect only a few positions, my best guess was to add a dummy device on both circuit terminations at the hub location. However, this requires me to also add a `Site` for the central hub (not owned by us), etc. Has anyone found a good way to model this without making it too complex?

**Comment by jsenecal** [comment]
I usually split the "Circuit" by fiber pairs or ranges of fibers and connect them accordingly to the rear-ports of the patch panel. 

This is also somewhat related to https://github.com/netbox-community/netbox/issues/13086 which is not exactly for the same outcome and usecase but could "fix" your case as well by allowing "child circuits"

**Reply by alehaa** [reply]
I‘ not sure whether this solves my issue. The circuit is not virtual, it just has positions like rear ports have already. So a Circuit might neer to have or understand positions, too. Fot interconnection, circuits would need to be connected directly and maybe just subsets of the positions. I‘ll give it a thought and maybe submit a FR.
